functions/HttpTriggerAPIUsersId/__init__.py

```diff
@@ -23,30 +32,48 @@ def main(req: func.HttpRequest) -> func.HttpResponse:
     id = req.route_params.get('userId')
     if not id:
         # If no userId is supplied http response -> Bad Request
+        logging.debug("User Id was not found")
         return func.HttpResponse(
             "Bad Request",
             status_code=400
         )
     else:
         try:
+            # Create a new connection
+            logging.debug("Attempting DB connection!")
             with pypyodbc.connect(connectionString) as conn:
-                return get_user_by_id(conn, id)
-        except pypyodbc.DatabaseError as e:
-            # This code is returned by pypyodbc meaning Unautroized when a bad password is supplied etc
-            # Adddtional error handeling will need to be implemented, maybe in srpint 2
-            if e.args[0] == '28000':
-                return func.HttpResponse(
-                    "Unauthorized",
-                    status_code=403
-                )
+                logging.debug("Connection to DB successful!")
+
+                if req.method == 'GET':
+                    return get(conn, id)
+                elif req.method == 'POST':
+                    return post()
+                elif req.method == 'PUT':
+                    return put(req, conn, id)
+                elif req.method == 'PATCH':
+                    return patch()
+                elif req.method == 'DELETE':
+                    return delete(req, conn, id)
 
+                conn.close()
+                logging.debug('Connection to DB closed')
 
-def get_user_by_id(conn, id):
+        except Exception as e:
```

Comment:
This blanket catch for exceptions should basically work, but doesn't really give us a lot of (in-code) information about what kinds of errors could arise. Comments would about this would be nice but you could also do a separate except for known errors?

Reply:
That is a good point, but I was focusing on catching all errors regardless of pyodbc or anything else. I could implement better error handling. But in this part, any error that occurs should be a 500 error.